### ISSUE TITLE
v2.0.0 plugin starts in error state

### ISSUE THREAD

**Issue by jonjesse1** [bitfocus/companion-module-magewell-proconvert-decoder#6]
Everytime Companion starts (companion-win64-2.4.0+4695-beta-42de8ba1.exe) some, but not all of my magewell decoder connections start in an error condition. 
`22.09.22 07:06:11 instance(226-RX-A): Error: TypeError: Cannot read property '0' of undefined`
This is remedied by hitting the edit button on the connection and immediately hitting save. The only properties defined in the connection are the decoder IP, username, pass, 1000ms polling interval, polling on, and the same thing happens with verbose logging on or off. verbose logging provides no additional info in the companion log file. 

this occurs on every restart of companion.

**Comment by cpuguru07** [comment]
Can confirm this is still happening in 2.4.0 stable (2.4.0+4877-v2-4.0-cd0b68c6). I have 5 instances of this module and it causes Companion to slow down/crawl. If you catch it right when Companion starts, you can go in and edit each instance and hit Save and it's fine, but if you don't catch it right away, it's hung so bad that it's almost impossible to edit/save the instances. Lots of errors relating to this module in the log, but I didn't save them before I cleared the log (sick brain is a little slow on the draw this morning).

**Comment by cpuguru07** [comment]
Still happening in 2.4.1 stable. I had to downgrade back to 2.3.1 because it kept causing Companion to hang/crash.

**Comment by cpuguru07** [comment]
Still an issue in 2.4.2. I'm stuck on 2.3.1 because newer versions of this module are unusable.
Has this module been abandoned?
@josephdadams @Julusian @JejeSwag

**Comment by josephdadams** [comment]
I can take a look at it. Thanks for tagging me in this, because this is my first time seeing it.

**Comment by cpuguru07** [comment]
@josephdadams Any update on this?

**Comment by josephdadams** [comment]
No. I have not had any free time. The module will also have to be upgraded to the companion v3 API.

**Comment by cpuguru07** [comment]
@josephdadams Is it possible to roll this module back to a version that didn't have these issues? I am unable to update to newer versions of Companion because this module makes all of Companion unusable.

**Comment by josephdadams** [comment]
What version of the module works for you?

**Comment by cpuguru07** [comment]
@josephdadams Whatever version is bundled with Companion v2.3.1 - appears to be module v1.0.6.

**Comment by josephdadams** [comment]
Does this happen if you start with a fresh install of Companion and config? i.e. setting up the connection new? Or only for existing connections

**Comment by cpuguru07** [comment]
@josephdadams
I reset my Companion configuration, but then I noticed there are still some "phantom" instances of this module.
See screenshot:
![image](https://user-images.githubusercontent.com/8798041/228653838-409894d8-2547-42c4-b9db-6d98a7f6cfba.png)
I wonder if this is contributing to the glitchiness of this module?

I'm going to attempt to uninstall Companion/delete the database and then reinstall.

**Comment by cpuguru07** [comment]
Clarification: Module will start (no longer causes Companion to crash).
The issue is with erratic behavior - instances crash, can no longer assign NDI sources, and module feedback variables are unstable.

Upon fresh install of Companion, the first instance of this module seems to behave correctly. However, upon adding a second/subsequent instances, they all experience these issues. The issue appears to be interference between multiple instances of this module. Even if I disable subsequent instances, the issues remain. The only way to get the module stable is to delete subsequent instances and only have one instance.
